File: common/src/main/java/com/hartwig/actin/molecular/datamodel/immunology/HlaAllele.java
package com.hartwig.actin.molecular.datamodel.immunology;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class HlaAllele {

    @NotNull
    public abstract String name();

    public abstract double tumorCopyNumber();

    public abstract boolean hasSomaticMutations();
}

File: common/src/main/kotlin/com/hartwig/actin/molecular/datamodel/orange/immunology/HlaAllele.kt
package com.hartwig.actin.molecular.datamodel.orange.immunology

data class HlaAllele(
    val name: String,
    val tumorCopyNumber: Double,
    val hasSomaticMutations: Boolean
) : Comparable<HlaAllele> {

    override fun compareTo(other: HlaAllele): Int {
        return name.compareTo(other.name)
    }
}
